python-sdk/tutorials/automl-with-azureml/forecasting-pipelines/scripts/infer.py

```diff
@@ -0,0 +1,210 @@
+import argparse
+import os
+import uuid
+import numpy as np
+import pandas as pd
+
+from pandas.tseries.frequencies import to_offset
+from sklearn.externals import joblib
+from sklearn.metrics import mean_absolute_error, mean_squared_error
+
+from azureml.data.dataset_factory import TabularDatasetFactory
+from azureml.automl.runtime.shared.score import scoring, constants as metrics_constants
+import azureml.automl.core.shared.constants as constants
+from azureml.core import Run, Dataset, Model
+
+try:
+    import torch
+
+    _torch_present = True
+except ImportError:
+    _torch_present = False
+
+
+def infer_forecasting_dataset_tcn(
+    X_test,
+    y_test,
+    model,
+    target_column_name,
+    max_horizon,
+    time_column_name,
+    output_dataset_name="results",
+):
+
+    if max_horizon is None or time_column_name is None:
+        print("Max Horizon or Time Column Name is None. Cannot infer.")
+        exit()
+
+    y_pred, df_all = model.forecast(X_test, y_test)
+
+    pd.merge(X_test, df_all)
+    run = Run.get_context()
+
+    registered_train = TabularDatasetFactory.register_pandas_dataframe(
+        df_all,
+        target=(run.experiment.workspace.get_default_datastore(), str(uuid.uuid4())),
```

Comment:
May be we can add forecast date instead of uuid here as we are running forecasts by schedule, so that user will understand, when the forecast was ran?

Reply:
changed